src/main/java/com/pnu/dev/shpaltaif/service/PostServiceImpl.java

```diff
@@ -136,7 +139,7 @@ public void deactivate(User user, Long id) {
     }
 
     private void checkUserAccessPermissions(Post post, User user) {
-        if (user.getRole() == UserRole.ROLE_ADMIN) {
+        if (user.getRole() == UserRole.ROLE_ADMIN || user.getRole() == UserRole.ROLE_EDITOR) {
```

Comment:
Do not change this and `delete` methods.
Add another method `checkUserEditOnlyAccessPermissions()`